bfd/elfnn-riscv.c

```diff
@@ -1878,10 +1899,9 @@ riscv_elf_relocate_section (bfd *output_bfd,
 	      && (!bfd_link_pic (info) || h->plt.offset == MINUS_ONE))
 	    {
 	      /* We can use x0 as the base register.  */
-	      bfd_vma insn = bfd_get_32 (input_bfd,
-					 contents + rel->r_offset + 4);
+	      bfd_vma insn = bfd_getl32 (contents + rel->r_offset + 4);
 	      insn &= ~(OP_MASK_RS1 << OP_SH_RS1);
-	      bfd_put_32 (input_bfd, insn, contents + rel->r_offset + 4);
+	      bfd_putl32 (insn, contents + rel->r_offset + 4);
```

Comment:
Oh I see you also use getl/putl in the gas/config/tc-riscv.c.  You can move your riscv_[get|put]_insn to bfd/elfxx-riscv.c, then assembler and dis-assembler can also use your functions/macros.  Besides, we probably need to get the instructions by get16l/get32l when dis-assembling, too.

Reply:
My thinking here is that `bfd_{get,put}lNN` is always correct, and that `riscv_{get,put}_insn` is only needed because there is no equivalent to `bfd_{get,put}` (which takes the number of bits as a parameter) which is explicitly little endian.  But maybe that logic becomes a little muddled when we get to 48-bit instructions and can't use `bfd_{get,put}l48` because that doesn't exist either...

I can move the macros to bfd/elfxx-riscv.h and use them more consistently if you like.

Disassembly uses `bfd_getl16` on the indiviual parcels, and shift-ors them together, which should be fine.  Trying to adapt `riscv_get_insn` to cover this use-case doesn't seem worthwile.

Reply:
I'm fine with this.  Just worried that there might be confused when we want to get/put the instructions, but there are two choices which are doing the same things.  Ideally, always use the same interfaces can reduce the burdens when developing.  But currently it should be fine, we can delay the discussion until we have more cases need to be considered.